### ISSUE TITLE
Reg\Register an Operation - External User can select "Registration Purpose\Electricity Import Operation" to see a simplified version of registration

### ISSUE THREAD

**Issue by andrea-williams** [bcgov/cas-registration#1962]
#### Dependencies

- ~Blocked by figuring out what to do if the user changes purpose?~
- ~Blocked by #1694~
- Blocking

#### Description:

EIOs are a somewhat unique type of operation in the sense that they don't have the concept of a facility, so we need to modify the registration workflow to suit them.

#### Acceptance Criteria:

Given I am an authenticated external user 
When I select "Electricity Import Operation" as my registration purpose (step 1)
Then I see a simplified version of the registration workflow with only the data fields that are relevant to EIOs
(a stripped-down version of the Operation form, no Facility form at all)

#### Development Checklist:

- [ ] [EIO wireframe for Operation section of form](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18034-12534&t=HA6SzABkE1DQWPQg-4)
- [x] depending on the add operation feature has previously been implemented, either create a new json form schema for EIOs and use it conditionally, or add conditional fields to the existing one
- [x] update the `Facility` db model to have a "Electricity Import" type
- [x] wherever necessary, update `Operation` and `Facility` db models to allow `null`s for the fields that don't apply to EIOs. 
- [x] on form submit:
-- [ ] new Operation record is created in the db for the EIO (submit button already calls the POST endpoint but may need to tweak to accommodate EIO)
-- [ ] remove "Facility Information" step in the registration workflow UI (see how new entrant/opt-in steps are conditionally added/removed as an example)
-- [ ] but a new Facility is still created in our database for the EIO (facility name should be set to the operation's name; facility type is "Electricity Import"; facility address should be set to the operation's address; lat/long of largest emissions can be empty (set to optional in db)
- [x] To still ensure all necessary data is provided, create new django-ninja schemas for `ElectricityImportOperationIn` and `ElectricityImportOperationOut` (don't make a new endpoint, use custom validation, see comment below)
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

**Comment by Sepehr-Sobhani** [comment]
I was not involved, and I am not quite sure about the first TBD, but for the second one, I think we can adjust our schemas and add some custom validation logic to enforce required fields based on the purpose.

**Comment by andrea-williams** [comment]
for the first TBD (`but a new Facility is still created in our database for the EIO (facility name should be set to the operation's name; facility type is "Electricity Import"; facility address should be set to the operation's address; lat/long of largest emissions - TBD)`) - this may be a question for the business area. Possible solutions I can think of are:

- we ask the external user to provide lat/long coordinates for their operation in addition to their address
- we make the lat/long coordinates optional in the database so that the "facility" for the EIO just doesn't have lat/long coordinates associated with it
- we use something like the Google Maps API (other brands are available) to fetch lat/long coordinates for a provided address. 

cc'ing @hannavovk and @zoeyli-46

**Comment by patrickisaac** [comment]
Aug. 13 Refinement Notes:
- Need additional information/refinement with developers
- Hanna to seek additional information from BA

**Comment by hannavovk** [comment]
Confirmed with business area that lat/long coordinates are not needed for EIOs. BA is in agreement with making lat/long coordinates optional in the database so that the "facility" for the EIO just doesn't have them, as long as the SFOs and LFOs still have to enter them.
(cc @andrea-williams, @zoeyli-46 )

**Comment by patriciarussellCAS** [comment]
I don't think this is blocked anymore, correct? It would be good to get this one in before Prod launch if possible :)

**Comment by andrea-williams** [comment]
> I don't think this is blocked anymore, correct? It would be good to get this one in before Prod launch if possible :)

I agree, not blocked anymore, and would be good to include in MVP.

**Comment by BCerki** [comment]
Testing instructions, new EIO
- go to the operations grid and click the add and register button
- choose EIO as the purpose and fill out the fields
- on the next page, copy the id in url (the thing that starts with c490: 

![Image](https://github.com/user-attachments/assets/9bf6b458-72d6-403d-be1b-1abc2976e0d3)). Because we can't see new operations in the admin grid at the moment, you'll use this id to get to the operation once you've finished creating it
- finish the registration flow
- type in the url `administration/operations/[the id you copied]`
- you should see the details for your new operation

Instructions for existing EIO:
- navigate to registration
- choose Blight EIO - Draft
- carry on through the registration flow

**Comment by patriciarussellCAS** [comment]
Hey @TammieV-BCG - just flagging that we won't be able to test EIO things in registration until this ticket is complete. So this might delay that UAT ticket.

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://bcgov-my.sharepoint.com/:w:/g/personal/francesca_viduya_gov_bc_ca/EQOBYsJZ9IRIsfywtqSCG0kBjBxVvBzs6Bv0L2UcvRd0Uw?e=0tgPoG).

I can verify that:
- [x] When I select "Electricity Import Operation" as registration purpose, I see a simplified version of the registration workflow with only the data fields relevant to EIOs as specified in [this wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18034-12534&t=HA6SzABkE1DQWPQg-4)
- [x] When I navigate to Dashboard > Operations, I see the new EIO created
- [x] In the Operations grid, the new EIO created has a value of ```N/A``` in the Facilities column  

The wireframe and the app does not exactly match but the fields do, moving to Pending PO for review of notes and test doc. Thank you!

**Comment by patriciarussellCAS** [comment]
Thanks! Test document reviewed. Moving to Accepted.
